/*
    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        https://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

syntax = "proto3";
import "google/protobuf/any.proto";
import "SerializableOperator.proto";
package NES;

option java_package = "stream.nebula.protobuf";
option java_multiple_files = true;

/*
The serializable wrapper definition for query plan
 */
message SerializableQueryPlan{
  map<uint64, SerializableOperator> operatorMap = 1;
  repeated uint64 rootOperatorIds = 2;
  optional uint64 queryId = 3;
}

message SubmitQueryRequest {
  SerializableQueryPlan queryPlan = 1;
  map<string, google.protobuf.Any> context = 2;
  optional string queryString = 3;
}